### PR TITLE
Environment_Engine: Create openings from a panel

### LINKED ISSUE #2595: Environment_Engine: Glazing panel type
#### Description:

Currently, BH.oM.Environment.Elements.PanelType does not allow for a panel which is entirely glazing. (The current workaround is to create an opening in a panel equal to the dimensions of the panel.) 

It should be possible to define a panel type as glazing. 

Additionally, I intend to use BH.Engine.Environment.Modify.SetPanelTypeByIntersectingLines in conjunction with this panel type, so the implementation needs to be compatible with this method.

### PR DESCRIPTION
**Pull request by robinflyman**
Closes #2595 

Created a method to create openings from a panel.

### CODE REVIEW

**Comment by robinflyman** [comment]
@BHoMBot check required

File: Environment_Engine/Convert/OpeningFromPanel.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using BH.oM.Reflection.Attributes;
```
Comment (FraserGreenroyd) [inline Environment_Engine/Convert/OpeningFromPanel.cs:25]:
```suggestion
using BH.oM.Base.Attributes;
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

File: Environment_Engine/Convert/OpeningFromPanel.cs (hunk 1 above)
Comment (FraserGreenroyd) [inline Environment_Engine/Convert/OpeningFromPanel.cs:25]:
```suggestion
using BH.oM.Base.Attributes;
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]

File: Environment_Engine/Convert/OpeningFromPanel.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Environment_Engine/Convert/OpeningFromPanel.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
Final commit was only to copyright, all checks had passed that were required on the previous commit. Am therefore granting neutral checks to avoid over using bot resource.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting a neutral check on: core, versioning, installer, project-compliance, ready-to-merge, documentation-compliance, code-compliance, null-handling, dataset-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: serialisation

**Pull request merged**